tests/helpers_tests/test_event_tag_utils.py

```diff
@@ -116,75 +127,265 @@ def test_get_revenue_value__revenue_tag_zero(self):
-    self.assertIsNone(event_tag_utils.get_numeric_value(0.5, logger=logger.SimpleLogger()))
-    self.assertIsNone(event_tag_utils.get_numeric_value(65536, logger=logger.SimpleLogger()))
-    self.assertIsNone(event_tag_utils.get_numeric_value(9223372036854775807, logger=logger.SimpleLogger()))
-    self.assertIsNone(event_tag_utils.get_numeric_value('9223372036854775807', logger=logger.SimpleLogger()))
-    self.assertIsNone(event_tag_utils.get_numeric_value(True, logger=logger.SimpleLogger()))
-    self.assertIsNone(event_tag_utils.get_numeric_value(False, logger=logger.SimpleLogger()))
+    with mock.patch('optimizely.logger.SimpleLogger.log') as mock_logger:
+        self.assertIsNone(event_tag_utils.get_numeric_value(None, logger=logger.SimpleLogger()))
+    calls = [
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is undefined.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        )
+    ]
+    mock_logger.assert_has_calls(calls)
+
+    calls = [
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        ),
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        ),
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        ),
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        ),
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        ),
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        ),
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
+        mock.call(
+            enums.LogLevels.WARNING,
+            'The provided numeric metric value None is in an invalid format and will not be sent to results.'
+        ),
+        mock.call(enums.LogLevels.DEBUG, 'Event tags is not a dictionary.'),
```

Comment:
You can easily avoid all this repetition by assigning string or mock.call values to a variable